reference-implementation/lib/abstract-ops/readable-streams.js

```diff
@@ -228,7 +235,8 @@ function ReadableStreamPipeTo(source, dest, preventClose, preventAbort, preventC
     });
 
     // Errors must be propagated backward
-    isOrBecomesErrored(dest, writer._closedPromise, storedError => {
+    WritableStreamDefaultWriterIsOrBecomesErrored(writer, () => {
```

Comment:
This new helper allows attaching a *synchronous* callback for when *dest* becomes `"erroring"` or `"errored"`, following the discussion in https://github.com/whatwg/streams/issues/1207#issuecomment-1017947644. I added a test for this in https://github.com/web-platform-tests/wpt/commit/1646d657bf258f850afb5f22860ec02dccbc6b7a.